app/routes/admin.js

```diff
@@ -0,0 +1,69 @@
+const router = require("express").Router();
+const mongoose = require("mongoose");
+const Request = require("../../models/request");
+const User = require("../../models/users");
+const Donor = require("../../models/donor");
+const bodyParser = require("body-parser");
+
+const authCheck = (req, res, next) => {
+  if (!req.user) {
+    res.redirect("/auth/google");
+  } else {
+    next();
+  }
+};
+
+router.get("/", authCheck, (req, res) => {
+  console.log(req.user);
+  Request.find().then(currentRequest => {
+    console.log(currentRequest);
+    var details = {
+      user_details: req.user,
+      request_details: currentRequest,
+      no_of_requests: currentRequest.length
+    };
+    console.log(details);
+    res.render("admin", { user: details });
+  });
+});
+
+router.get("/donor/add", (req, res) => {
+  var mobileNumber = req.body.mobilenumber;
+  var bloodGroup = req.body.bloodgroup;
+  var donorName = req.body.donorname;
+  new Donor({
+    name: donorName,
+    mobileNumber: mobileNumber,
+    bloodGroup: bloodGroup
+  })
+    .save()
+    .then(currentDonor => {
+      console.log(currentDonor);
+    });
+  res.redirect("/admin");
+});
+
+router.post("/status/change", (req, res) => {
```

Comment:
can be /request/approve, coz only two status is there ri8, pending and approved.

Reply:
also have reject route too

Reply:
/status/change will change the status either to approved or rejected, or the user who created the request can even cancel the request. so the values can be approved,rejected,pending,cancelled.